.github/workflows/integration.yml

```diff
@@ -141,6 +141,8 @@ jobs:
     if: ${{ github.ref == 'refs/heads/master' }}
     needs: [unit-and-coverage-tests, functional-tests, check-commit-message]
     runs-on: ubuntu-latest
+    permissions:
+        id-token: write
```

Comment:
Is there any reason we need to elevate permission on `id-token`? According to [`npm provenance` documentation](https://docs.npmjs.com/generating-provenance-statements), this seems mandatory to authenticate the action with its `GITHUB_TOKEN`. But can't find a documentation of the behavior of `id_token`.

Reply:
I didn't get into the details of the id-token permission, I followed the npm provenance doc that said to add it. You can find the documentation about id-token [here](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs)

Reply:
Thanks! I understand the rationale now!

Ready To Merge then!